### PR TITLE
Trivial improvements for bit vector simplification

### PR DESCRIPTION
**Pull request by hzzhang**
This pull request implements the simplification of bit vector items in some trivial cases, such as simplifying BVAdd(x, BVZero(n)) to BVZero(n). Therefore, users may not need to call the expensive BDD simplifier for those trivial cases.

### CODE REVIEW

**Comment by hzzhang** [comment]
@marcogario 
The continuous-integration/appveyor/pr failed, and the details showed 

> AppVeyor failed to assign job to 'gce' build environment after 3 attempt(s). The envronment is either unavailable or there are no free workers, however alternate build environment was not configured by the system administrator. Please try again later.

Does it mean something goes wrong in the CI system itself?

File: pysmt/test/test_bv_simplification.py [hunk 1]
```diff
@@ -94,6 +95,316 @@ def all_special(self):
...
+        f = BVMul(x, BVZero(32))
+        self.assertEqual(f.simplify(), BVZero(32))
+
+    def test_bv_mul_1(self):
+        x = Symbol("x", BVType(32))
+        f = BVMul(x, BVOne(32))
+        self.assertEqual(f.simplify(), Symbol("x", BVType(32)))
+
+    def test_bv_mul_constants(self):
+        f = BVMul(BV(10, 32), BV(12, 32))
```
Comment (marcogario) [inline pysmt/test/test_bv_simplification.py]: Does this consider overflow?

File: pysmt/test/test_bv_simplification.py [hunk 2]
```diff
@@ -94,6 +95,316 @@ def all_special(self):
                 for op in self.special_operators:
                     self.check(op(l, n))
 
+    def test_bv_0_add(self):
+        x = Symbol("x", BVType(32))
+        f = BVAdd(BVZero(32), x)
+        self.assertEqual(f.simplify(), Symbol("x", BVType(32)))
+
```
Comment (marcogario) [inline pysmt/test/test_bv_simplification.py]: I would use directly x in the assert. It is more clear.

File: pysmt/test/test_bv_simplification.py [hunk 3]
```diff
@@ -94,6 +95,316 @@ def all_special(self):
...
+        f = BVUDiv(x, BVZero(32))
+        self.assertEqual(f.simplify(), BV(2**32 - 1, 32))
+
+    def test_bv_0_udiv_0(self):
+        f = BVUDiv(BVZero(32), BVZero(32))
+        self.assertEqual(f.simplify(), BV(2**32 - 1, 32))
+
+    def test_bv_nonzero_udiv_0(self):
+        f = BVUDiv(BV(10, 32), BVZero(32))
+        self.assertEqual(f.simplify(), BV(2**32 - 1, 32))
```
Comment (marcogario) [inline pysmt/test/test_bv_simplification.py]: I don't remember what is the agreed value for this. This is undefined in arithmetic theories. Is this defined in the smtlib standard?

**Comment by marcogario** [comment]
For the ci, I think mikand ia working on this in anothet branch. If travis is green, that is good for me.

File: pysmt/test/test_bv_simplification.py (hunk 1 above)
Comment (hzzhang) [reply pysmt/test/test_bv_simplification.py]: No, but I'll add tests for overflow cases in the next PR.

File: pysmt/test/test_bv_simplification.py (hunk 2 above)
Comment (hzzhang) [reply pysmt/test/test_bv_simplification.py]: Will change in the next PR.

File: pysmt/test/test_bv_simplification.py (hunk 3 above)
Comment (hzzhang) [reply pysmt/test/test_bv_simplification.py]: Simplifier::walk_bv_udiv() before this PR does so, as well as Z3, though the bit vector division by zero seems invalid in [smtlib standard](http://smtlib.cs.uiowa.edu/theories-FixedSizeBitVectors.shtml):

>   o _ div _,  which takes an integer x ≥ 0 and an integer y > 0 and returns
>     the integer part of x divided by y (i.e., truncated integer division).
>
>   o _ rem _, which takes an integer x ≥ 0 and y > 0 and returns the
>     remainder when x is divided by y.  Note that we always have the following
>     equivalence for y > 0: (x div y) * y + (x rem y) = x.

**Comment by hzzhang** [comment]
@marcogario 
I've updated the PR and CI now failed at test_tactics_z3. The test case failed due to a segment fault, which I cannot reproduce locally. It succeeded before the update, and all updates are in test_bv_simplification.py and should not affect test_tactics_z3. Does it relate to #465 which may not be completely fixed by #530 ?

**Comment by mikand** [comment]
@hzzhang I just re-launched the CI test, let's see if it it deterministic or not. Interestingly, this time the segfauult happened on OSX, while in #465 it only happened on Linux...

**Comment by hzzhang** [comment]
@mikand All tests passed at this time.
